src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/ThirdPartyLicenseResolver.cs

```diff
@@ -7,32 +6,59 @@ namespace DotNetProjectFile.Analyzers.MsBuild;
 [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
 public sealed class ThirdPartyLicenseResolver() : MsBuildProjectFileAnalyzer(
     Rule.OnlyIncludePackagesWithExplicitLicense,
-    Rule.PackageOnlyContainsDeprecatedLicenseUrl)
+    Rule.PackageOnlyContainsDeprecatedLicenseUrl,
+    Rule.PackageContainsIncompatibleLicense)
 {
     /// <inheritdoc />
     public override bool DisableOnFailingImport => false;
 
     /// <inheritdoc />
-    protected override void Register(ProjectFileAnalysisContext<MsBuildProject> context)
+    protected override void Register(ProjectFileAnalysisContext context)
     {
+        var allowed = (context.GetMsBuildProperty("AllowedLicenses") ?? string.Empty)
```

Comment:
Shouldn't this default to `Licenses.All` if no explicit allow list has been provided? That way it just assumes anything is fine as long as the licenses don't violate each other.

Reply:
I think we should enforce this, and it would be fine to (if not specified) allow all permissive licenses (we might even want to have a single expression that contains all together) if not specified otherwise, but reject copy-left licenses unless explicitly allowed.

Reply:
But if you're own license is AGPL3 or GPL3 there is not really much point in requiring people to manually add GPL3 to their allow-list in my opinion, since the compatibility is already known. I would expect manually setting the list of allowed licenses is mostly useful for companies that only allow a certain pre-verified list of licenses to be used

Reply:
That is fair. As most projects are not packages their own, I suggest to only implement that scenario for now, and in a follow-up take the compatibility of the project itself into account. The more I think of it, the more I think the logic and communication to the user benefits from having a different rule stating the the package has a license that conflicts the license of the project.